app.js

```diff
@@ -0,0 +1,135 @@
+'use strict';
+
+// global variables
+var votesRemaining = 25;
+var picturesEl = document.getElementById('pictures');
+var resultsEl = document.getElementById('results');
+var picOneEl = document.getElementById('pic-one');
+var picTwoEl = document.getElementById('pic-two');
+var picThreeEl = document.getElementById('pic-three');
+var containerEl = document.getElementById('pictureContainer');
+var allProductArr = [];
+var recentIndex = [];
+
+function Product (filePath, productName){
+    this.alt = productName;
+    this.title = productName;
+    this.src = filePath;
+    this.votes = 0;
+    this.views = 0;
+
+    allProductArr.push(this);
+}
+// pulls random image from the constructor array
+function imageGenerator(){
+    
+    while(recentIndex.length > 6){
+        recentIndex.shift();
+    }
+    console.log( 'recent index', recentIndex);
+
+    var index = random(allProductArr.length);
+
+    while(recentIndex.includes(index)) {
+        index = random(allProductArr.length);
+    }
+    picOneEl.src = allProductArr[index].src;
+    picOneEl.alt = allProductArr[index].alt;
+    picOneEl.title = allProductArr[index].title;
+    allProductArr[index].views++;
+    recentIndex.push(index);
+
+
+    var indexTwo = random(allProductArr.length);
```

Comment:
Once you have everything working how you want, I think a cool way to refactor this would be to call this method 3 times instead of have it run similar code 3 times. Again, not something to focus on now!